tests/bearlib/DeprecateSettingsTest.py

```diff
@@ -10,8 +11,22 @@ def func(new):
     """
 
 
+@deprecate_settings(x=('a', lambda a: a+1), y=('a', lambda a: a+2))
+def func_2(x, y):
+    return x+y
+
+
 class DeprecateSettingsTest(unittest.TestCase):
 
     def test_docstring(self):
         self.assertEqual(func.__doc__.strip(),
                          'This docstring will not be lost.')
+
+    def test_multiple_args(self):
+        logger = logging.getLogger()
+
+        with self.assertLogs(logger, 'WARNING') as cm:
+            self.assertEqual(func_2(a=1), 5)
+
+            self.assertRegex(cm.output[0], 'WARNING:root:The setting `a` is '
+                             r'deprecated. Please use `[xy]` instead.')
```

Comment:
> Please use `[xy]` instead

This needs to be improved.

And we also need a test to cover the other scenario:

> 'The value of `{}` and `{}` are conflicting. `{}` will be used instead.

Reply:
> > Please use [xy] instead
>
> This needs to be improved.

Not sure if you/I understand correctly, but this is a regex, so the actual error is "Please use `x` instead" or "Please use `y` instead". The reason this needs a regex is that the order of args seems to depend on the version of the py(py3.4 has the message with y, py3.5 with x).

Reply:
> And we also need a test to cover the other scenario:
>
> > 'The value of {} and {} are conflicting. {} will be used instead.

Added.

Reply:
No, there's only a single error message. This seems to be a deliberate choice in #3204.